src/pretix/plugins/sendmail/views.py

```diff
@@ -36,13 +39,35 @@ def form_valid(self, form):
         self.request.event.log_action('pretix.plugins.sendmail.sent', user=self.request.user, data=dict(
             form.cleaned_data))
 
+        tz = pytz.timezone(self.request.event.settings.timezone)
+
         failures = []
+        self.output = {}
         for o in orders:
-            try:
-                mail(o.email, form.cleaned_data['subject'], form.cleaned_data['message'],
-                     None, self.request.event, locale=o.locale, order=o)
-            except SendMailException:
-                failures.append(o.email)
+            if self.request.POST.get("action") == "preview":
+                self.locales_list = self.request.event.settings.locales
+                for l in self.locales_list:
```

Comment:
There's no need for the class variable, ``for l in self.request.event.settings.locales`` is totally fine.